File: ai-service/routes/mood_match_routes.py
from flask import Blueprint, request, jsonify
from services.mood_match_service import MoodMatchService

mood_match_blueprint = Blueprint('mood_match', __name__)
service = MoodMatchService()

@mood_match_blueprint.route('/mood-match', methods=['POST'])
def handle_mood_match():
    data = request.get_json()
    if not data:
        return jsonify({'error': 'Bad Request', 'message': 'Missing JSON body'}), 400
        
    if 'mood_query' not in data or 'catalog' not in data:
        return jsonify({'error': 'Bad Request', 'message': 'Missing required fields: mood_query, catalog'}), 400
        
    result = service.match_mood(data['mood_query'], data['catalog'])
    return jsonify({'status': 'success', 'data': result}), 200

File: src/routes/moodMatch.routes.js
const express = require('express');
const moodMatchService = require('../services/moodMatch.service');
const { authenticate } = require('../middleware/auth');
const { aiLimiter } = require('../middleware/rateLimiter');
const { buildResponse } = require('../utils/helpers');

const router = express.Router();

router.post('/', authenticate, aiLimiter, async (req, res, next) => {
  try {
    if (!req.body.mood_query) {
      return res.status(400).json(buildResponse(false, null, 'mood_query is required'));
    }
    const result = await moodMatchService.matchMood(req.body.mood_query);
    res.status(200).json(buildResponse(true, result, 'Mood matched successfully'));
  } catch (err) {
    next(err);
  }
});

module.exports = router;
